spec/search_spec.js

```diff
@@ -298,11 +298,12 @@ describe('search', () => {
         scrollToView(scheduleFilter);
         scheduleFilter.click();
 
-        var scheduleDay = element(by.buttonText('Last 24 Hours'));
+        element.all(by.buttonText('Last 24 Hours')).then((items) => {
+            expect(items.length).toBe(4);
+            scrollToView(items[3]);
+            items[3].click();
```

Comment:
That's a very error-prone approach to click the 3rd item. Tests will break if you swap the order of filters or add another filter while it is easy way to avoid it. To do that first get the group of filters you are testing which is probably 'Date scheduled' and then search for 'Last 24 Hours' in that group. There will only be one match this way.